facade/src/main/java/org/jboss/pnc/facade/providers/ProductVersionProviderImpl.java

```diff
@@ -158,6 +164,25 @@ private void validateGroupConfigsBeforeUpdating(String id, ProductVersion restEn
         }
     }
 
+    private void validateMilestone(String id, ProductVersion entity) {
+        if (entity.getCurrentProductMilestone() != null) {
+            Integer newMilestoneId = Integer.parseInt(entity.getCurrentProductMilestone().getId());
+            if (newMilestoneId != null) {
+                org.jboss.pnc.model.ProductVersion productVersion = repository.queryById(Integer.parseInt(id));
```

Comment:
use Product version mapper method to get id mapper to map the product version id.